### ISSUE TITLE
cannot compile .xtx files, but can compile same file with .tex extension 

### ISSUE THREAD

**Issue by alexdetrano** [SublimeText/LaTeXTools#1246]
Hi,
I'm having a strange issue using LatexTools. I cannot compile files using xelatex with the .xtx extension.  The strange thing is, if I change the file extension to .tex, the compilation works.  It's not an issue with my xelatex, since I can compile from the command line using xelatex.  Very strange.

Here is some information, let me know if I can provide anything else.
Thanks,
Alex

**hello_world.xtx**
```latex
\documentclass[]{article}
\begin{document}
Hello world!
\end{document}  
```
Running either "Latex - XeLaTeX" or "Latex - Basic - XeLaTeX" results in the same error. The log file from sublime is below.
```
[Compiling C:\Users\alexd\Dropbox\resume\github_issue2\hello_world.xtx]

TraditionalBuilder: Engine: xelatex. Invoking texify... done.

Errors:

C:\Users\alexd\Dropbox\resume\github_issue2\hello_world.xtx:1: Undefined control sequence. [\documentclass]
C:\Users\alexd\Dropbox\resume\github_issue2\hello_world.xtx:2: Undefined control sequence. [\begin]

No warnings.

[Done!]
```

And the .log file from a failed sublime run:
```
This is XeTeX, Version 3.14159265-2.6-0.99998 (MiKTeX 2.9.6350 64-bit) (preloaded format=xetex 2017.10.14)  14 OCT 2017 21:08
entering extended mode
**C:/Users/alexd/Dropbox/resume/github_issue2/hello_world.xtx
(C:/Users/alexd/Dropbox/resume/github_issue2/hello_world.xtx
! Undefined control sequence.
l.1 \documentclass
                  []{article}
The control sequence at the end of the top line
of your error message was never \def'ed. If you have
misspelled it (e.g., `\hobx'), type `I' and the correct
spelling (e.g., `I\hbox'). Otherwise just continue,
and I'll forget about whatever was undefined.

! Undefined control sequence.
l.2 \begin
          {document}
The control sequence at the end of the top line
of your error message was never \def'ed. If you have
misspelled it (e.g., `\hobx'), type `I' and the correct
spelling (e.g., `I\hbox'). Otherwise just continue,
and I'll forget about whatever was undefined.

[1] )
Output written on hello_world.pdf (1 page).

```
If I run xelatex manually, the compile works:
```bash
alexd@DESKTOP-BSMG180 ~/alexd/Dropbox/resume/github_issue2
$ xelatex hello_world.xtx
This is XeTeX, Version 3.14159265-2.6-0.99998 (MiKTeX 2.9.6350 64-bit)
entering extended mode
(hello_world.xtx
LaTeX2e <2017-04-15>
Babel <3.10> and hyphenation patterns for 75 language(s) loaded.
("C:\Program Files\MiKTeX 2.9\tex\latex\base\article.cls"
Document Class: article 2014/09/29 v1.4h Standard LaTeX document class
("C:\Program Files\MiKTeX 2.9\tex\latex\base\size10.clo"))
No file hello_world.aux.
[1] (hello_world.aux) )
Output written on hello_world.pdf (1 page).
Transcript written on hello_world.log.
```

In case you want to see hello_world.log from a successful command line run:
```
alexd@DESKTOP-BSMG180 ~/alexd/Dropbox/resume/github_issue2
$ cat hello_world.log
This is XeTeX, Version 3.14159265-2.6-0.99998 (MiKTeX 2.9.6350 64-bit) (preloaded format=xelatex 2017.10.13)  15 OCT 2017 05:04
entering extended mode
**./hello_world.xtx
(hello_world.xtx
LaTeX2e <2017-04-15>
Babel <3.10> and hyphenation patterns for 75 language(s) loaded.
("C:\Program Files\MiKTeX 2.9\tex\latex\base\article.cls"
Document Class: article 2014/09/29 v1.4h Standard LaTeX document class
("C:\Program Files\MiKTeX 2.9\tex\latex\base\size10.clo"
File: size10.clo 2014/09/29 v1.4h Standard LaTeX file (size option)
)
\c@part=\count79
\c@section=\count80
\c@subsection=\count81
\c@subsubsection=\count82
\c@paragraph=\count83
\c@subparagraph=\count84
\c@figure=\count85
\c@table=\count86
\abovecaptionskip=\skip41
\belowcaptionskip=\skip42
\bibindent=\dimen102
)
No file hello_world.aux.
\openout1 = `hello_world.aux'.

LaTeX Font Info:    Checking defaults for OML/cmm/m/it on input line 2.
LaTeX Font Info:    ... okay on input line 2.
LaTeX Font Info:    Checking defaults for T1/cmr/m/n on input line 2.
LaTeX Font Info:    ... okay on input line 2.
LaTeX Font Info:    Checking defaults for OT1/cmr/m/n on input line 2.
LaTeX Font Info:    ... okay on input line 2.
LaTeX Font Info:    Checking defaults for OMS/cmsy/m/n on input line 2.
LaTeX Font Info:    ... okay on input line 2.
LaTeX Font Info:    Checking defaults for TU/lmr/m/n on input line 2.
LaTeX Font Info:    ... okay on input line 2.
LaTeX Font Info:    Checking defaults for OMX/cmex/m/n on input line 2.
LaTeX Font Info:    ... okay on input line 2.
LaTeX Font Info:    Checking defaults for U/cmr/m/n on input line 2.
LaTeX Font Info:    ... okay on input line 2.
[1

] (hello_world.aux) )
Here is how much of TeX's memory you used:
 202 strings out of 427834
 2063 string characters out of 3152469
 60877 words of memory out of 3000000
 4247 multiletter control sequences out of 15000+200000
 3648 words of font info for 15 fonts, out of 3000000 for 9000
 1328 hyphenation exceptions out of 8191
 22i,4n,17p,137b,107s stack positions out of 5000i,500n,10000p,200000b,50000s

Output written on hello_world.pdf (1 page).

alexd@DESKTOP-BSMG180 ~/alexd/Dropbox/resume/github_issue2

```

**Comment by r-stein** [comment]
I think is is a MikTeX/Latexmk issue. You can just press ctrl+shift+b and use *Basic Builder - XeLaTeX* to directly use the xelatex command.

**Comment by r-stein** [comment]
PS. I see you said that *Latex - Basic - XeLaTeX* results in the same error. Does it also have the same error output?

**Comment by alexdetrano** [comment]
first I run Latex - Basic Builder - Xelatex and get the following output in sublime build:
```
[Compiling C:\Users\alexd\Dropbox\resume\github_issue3\hello_world_xtx.xtx]

Basic Builder: running xelatex...done.

Could not read log file hello_world_xtx.log

Output from compilation:

This is XeTeX, Version 3.14159265-2.6-0.99998 (MiKTeX 2.9.6350 64-bit)
entering extended mode
! I can't find file `hello_world_xtx'.
<*> hello_world_xtx
                   
Please type another input file name
! Emergency stop.
<*> hello_world_xtx
                   
No pages of output.
Transcript written on texput.log.

```
Next, I run

Latex - Xelatex causes the undefined control sequence error.

**Comment by r-stein** [comment]
You can try to add `".xdx"` to [`"tex_file_exts"`](https://github.com/SublimeText/LaTeXTools/blob/master/LaTeXTools.sublime-settings#L67)

**Comment by alexdetrano** [comment]
I already did that, actually I couldn't even use any build commands until I
added that setting

**Comment by r-stein** [comment]
Can you open the ST console ``ctrl+` `` and check which commands are executed during the build process (they should be logged by default)?

**Comment by alexdetrano** [comment]
```
<class '_latextools_traditionalBuilder.TraditionalBuilder'>
3
Welcome to thread Thread-12
['texify', '-b', '-p', '--engine=xetex', '--tex-option="--synctex=1"', 'hello_world_xtx.xtx']
Running "'C:\Program Files\MiKTeX 2.9\miktex\bin\x64\texify.EXE' -b -p --engine=xetex '--tex-option="--synctex=1"' hello_world_xtx.xtx"
Finished normally
1

```

**Comment by r-stein** [comment]
Please use the basic builder and then provide the console output

**Comment by alexdetrano** [comment]
oops, here is the basic:

```
<class '_latextools_basicBuilder.BasicBuilder'>
3
Welcome to thread Thread-13
['xelatex', '-interaction=nonstopmode', '-synctex=1', 'hello_world_xtx']
Running "'C:\Program Files\MiKTeX 2.9\miktex\bin\x64\xelatex.EXE' -interaction=nonstopmode -synctex=1 hello_world_xtx"
Finished normally
1
```

**Comment by r-stein** [comment]
It seems to strip the extension. I will have a look at it later.

**Comment by ig0774** [comment]
This should be fixed in v3.15.1.
